operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java

```diff
@@ -64,13 +89,36 @@ private void stopLeading() {
   }
 
   private String identity(LeaderElectionConfiguration config) {
-    String id = config.getIdentity().orElse(System.getenv("HOSTNAME"));
+    var id = config.getIdentity().orElse(System.getenv("HOSTNAME"));
     if (id == null || id.isBlank()) {
       id = UUID.randomUUID().toString();
     }
     return id;
   }
 
+  private static Optional<String> tryNamespaceFromPath() {
```

Comment:
I think we could reasonably ask for the related method to become public in the Fabric8 client so that we don't have to maintain it here…

Reply:
Yeah that would be better. Do you think I could give it a shot?

Reply:
I will give it a try. On the other hand, if we want to get this shipped, we don't need to get blocked because the roundtrip might take some time.

Reply:
yep, we can adjust it here after, and ship this as it is. Maybe just create an issue so we don't forget it.